### PR TITLE
feat: jump-to-bottom button

### PR DESCRIPTION
**Pull request by suprko**
#### Relevant info and issue/PR links 
 https://app.clubhouse.io/peerio/story/4850/desktop-jump-to-most-recent-message
https://github.com/PeerioTechnologies/peerio-icebear/pull/177

#### Testing instructions  
When user has a chat open + user has scrolled up any distance from the bottom + user receives new messages in that chat:
* jump-to-bottom button with unread message counter should appear just above message input
* jump-to-bottom button and unread count should persist until user has scrolled to bottom, either manually or by pressing the jump-to-bottom button

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

**Comment by seavan** [comment]
@suprko I think you can implement the remaining part in the SDK, too.

what you need to do:

- there's an observable in desktop, which prevents autoscrolling when the chat view is scrolled far enough away
https://github.com/PeerioTechnologies/peerio-desktop/blob/dev/src/ui/chat/components/MessageList.jsx#L128

- this observable should be moved to clientApp and reset on component MessagesList dismount

- when chat is changed, reset this flag:
https://github.com/PeerioTechnologies/peerio-icebear/blob/feat-app-label/src/models/chats/chat.js#L47

- when the flag is set, this reaction should not markAllAsSeen:

https://github.com/PeerioTechnologies/peerio-icebear/blob/feat-app-label/src/models/chats/chat.message-handler.js#L25

i think that should be enough to finish the task.

**Comment by suprko** [comment]
@seavan I've got this working. In your last comment you referenced things in `feat-app-label` branch which I can't view now 😆 

Here's what has been done:
* desktop MessageList `stickToBottom` has been moved to icebear clientApp `messageStickToBottom`
  * desktop changes the flag the same way as before
  * `messageStickToBottom=true` whenever a new MessageList is mounted (change chat)
* when `!messageStickToBottom`, icebear chat.message-handler does not `markAllAsRead`. unread count (and jump-to-bottom button) are persistent until users scrolls to bottom, either manually or with button

Nothing has changed in chat.js because I wasn't sure what flag is supposed to be reset...

**Comment by seavan** [comment]
@suprko feat-app-label was merged to dev.

checked code, looks good to me. passing on to testing.

File: src/ui/chat/ChatView.jsx [hunk 1]
```diff
@@ -240,6 +240,28 @@ class ChatView extends React.Component {
             <ChatSideBar open={uiStore.prefs.chatSideBarIsOpen} />;
     }
 
+    @observable messageListRef;
```
Comment (anri-asaturov) [inline src/ui/chat/ChatView.jsx]: But we already have refs and scrolling methods (including smooth scroll) inside MessageList, why did you duplicate it here?

File: src/ui/chat/components/MessageInput.jsx [hunk 2]
```diff
@@ -112,6 +113,31 @@ class MessageInput extends React.Component {
...
+    }
+
+    @computed get snackbarVisible() {
+        if (this.snackbarRef) return this.snackbarRef.isVisible;
+        return false;
+    }
+
+    renderJumpToBottom() {
+        const chat = chatStore.activeChat;
+        if (chat.unreadCount === 0 || clientApp.messageStickToBottom) return null;
```
Comment (anri-asaturov) [inline src/ui/chat/components/MessageInput.jsx]: Why don't we let user quickly go back to the last messages even if there are no new messages? I scrolled to far back in history and want to go to most recent, sounds like a reasonable case?

**Comment by anri-asaturov** [comment]
This is not ready to test yet

File: src/ui/chat/components/MessageInput.jsx (hunk 2 above)
Comment (suprko) [reply src/ui/chat/components/MessageInput.jsx]: 🤷‍♂️ ask design/Skylar

Comment (anri-asaturov) [reply src/ui/chat/components/MessageInput.jsx]: Pauls says it should be as i said

File: src/ui/chat/ChatView.jsx (hunk 1 above)
Comment (suprko) [reply src/ui/chat/ChatView.jsx]: The button itself needs to be placed in MessageInput for positioning reasons, whereas the scrolling method is in MessageList. So the scrolling method is passed to `MessageInput` as a prop in ChatView.

The issue is described in greater detail in this comment. tl;dr, we can put the button in MessageList itself, but then the positioning would need to calculate MessageInput height (which can increase with long messages).
https://app.clubhouse.io/peerio/story/4850/desktop-jump-to-most-recent-message#activity-5635

Comment (anri-asaturov) [reply src/ui/chat/ChatView.jsx]: okay

**Comment by ebilg** [comment]
looks good, can be merged

**Pull request merged**